api/database/factories/TrainingOpportunityFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Database\Factories;
+
+use App\Enums\CourseFormat;
+use App\Enums\CourseLanguage;
+use App\Models\TrainingOpportunity;
+use Illuminate\Database\Eloquent\Factories\Factory;
+
+class TrainingOpportunityFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = TrainingOpportunity::class;
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        $title = $this->faker->jobTitle();
+        $description = $this->faker->sentences(3, true);
+        $applicationUrl = $this->faker->url();
+        $registrationDeadline = $this->faker->dateTimeBetween('-6 months', '6 months');
+        $trainingStart = $this->faker->dateTimeBetween($registrationDeadline, '9 months');
+        $trainingEnd = $this->faker->boolean() ? $this->faker->dateTimeBetween($trainingStart, '12 months') : null;
```

Comment:
Not a big deal but faker has an [`optional()` modifier ](https://fakerphp.org/#modifiers:~:text=values%20per%20provider%0A%0A//-,optional,-()%20sometimes%20bypasses%20the) that does a similar thing.

Reply:
I kinda prefer the boolean, can easily modify it by passing in a weight value

Reply:
Turns out it does the same thing, changed in https://github.com/GCTC-NTGC/gc-digital-talent/pull/12010/commits/67bf75a928085e2ffd8a08b5278c5068ef3d8920